packages/transaction-controller/src/utils.ts

```diff
@@ -208,26 +207,3 @@ export function getAndFormatTransactionsForNonceTracker(
       };
     });
 }
-
-/**
- * Checks whether a given transaction matches the specified network or chain ID.
- * This function is used to determine if a transaction is relevant to the current network or chain.
- *
- * @param transaction - The transaction metadata to check.
- * @param chainId - The chain ID of the current network.
- * @param networkId - The network ID of the current network.
- * @returns A boolean value indicating whether the transaction matches the current network or chain ID.
- */
-export function transactionMatchesNetwork(
-  transaction: TransactionMeta,
-  chainId: Hex,
-  networkId: string | null,
-) {
-  if (transaction.chainId) {
-    return transaction.chainId === chainId;
-  }
-  if (transaction.networkID) {
-    return transaction.networkID === networkId;
-  }
-  return false;
-}
```

Comment:
Unsure how to migrate Transaction state away from networkId

See https://github.com/MetaMask/metamask-extension/pull/20652#issuecomment-1707351348